mysql-test/suite/parts/t/alter_data_directory_innodb.test

```diff
@@ -0,0 +1,34 @@
+--source include/have_innodb.inc
+--source include/have_partition.inc
+
+--echo #
+--echo # MDEV-15953 Alter InnoDB Partitioned Table Moves Files (which were originally not in the datadir) to the datadir
+--echo #
+
+mkdir $MYSQLTEST_VARDIR/tmp/partitions_here;
```

Comment:
The test should clean this up by executing the corresponding `rmdir` command. (Because InnoDB creates but does not remove an additional subdirectory in the directory, that one should be explicitly removed as well.)